### PR TITLE
DOC: PR06 docstring fixes

### PR DESCRIPTION
**Pull request by tolaa001**
- This PR contains doctrine PR06 fixes mostly doing the conversion below:

- boolean to bool

- string to str

- integer to int

- Tests unchanged
- black pandas ran successfully

### CODE REVIEW

File: pandas/core/arrays/categorical.py [hunk 1]
```diff
@@ -634,7 +634,7 @@ def from_codes(cls, codes, categories=None, ordered=None, dtype=None):
 
         Parameters
         ----------
-        codes : array-like, integers
+        codes : array-like, int
```
Comment (simonjayhawkins) [inline pandas/core/arrays/categorical.py]: not sure about this one. IMO not so clear. maybe array-like[int] or possibly Sequence[int]. @datapythonista ?

File: pandas/core/arrays/categorical.py [hunk 2]
```diff
@@ -645,7 +645,7 @@ def from_codes(cls, codes, categories=None, ordered=None, dtype=None):
             Whether or not this categorical is treated as an ordered
             categorical. If not given here or in `dtype`, the resulting
             categorical will be unordered.
-        dtype : CategoricalDtype or the string "category", optional
+        dtype : CategoricalDtype or the str "category", optional
```
Comment (simonjayhawkins) [inline pandas/core/arrays/categorical.py]:
```suggestion
        dtype : CategoricalDtype or {"category"}, optional
```

File: pandas/core/arrays/categorical.py (hunk 1 above)
Comment (datapythonista) [reply pandas/core/arrays/categorical.py]: we use `array-like of int` (or `list of int`...)

File: pandas/core/arrays/datetimelike.py [hunk 3]
```diff
@@ -728,7 +728,7 @@ def _maybe_mask_results(self, result, fill_value=iNaT, convert=None):
         ----------
         result : a ndarray
         fill_value : object, default iNaT
-        convert : string/dtype or None
+        convert : str/dtype or None
```
Comment (datapythonista) [inline pandas/core/arrays/datetimelike.py]: `str, dtype or None` would be better. Besides consistency, in the future I'd like to be able to parse the types.

**Review by datapythonista — COMMENTED** [review]
Looks good, thanks for working on those @tolaa001. Just added couple of comments.

File: pandas/core/arrays/categorical.py (hunk 2 above)
Comment (jorisvandenbossche) [reply pandas/core/arrays/categorical.py]: I think for a single option, the {} brackets look a bit strange, I would maybe just do `CategoricalDtype or "category"`

File: pandas/core/arrays/categorical.py [hunk 4]
```diff
@@ -647,8 +647,8 @@ def from_codes(cls, codes, categories=None, ordered=None, dtype=None):
             Whether or not this categorical is treated as an ordered
             categorical. If not given here or in `dtype`, the resulting
             categorical will be unordered.
-        dtype : CategoricalDtype or the string "category", optional
-            If :class:`CategoricalDtype`, cannot be used together with
+        dtype : CategoricalDtype or "category", optional
+            If :class:`Categoarray-like, intricalDtype`, cannot be used together with
```
Comment (simonjayhawkins) [inline pandas/core/arrays/categorical.py]: whoops!

Comment (tolaa001) [reply pandas/core/arrays/categorical.py]: thanks for catching that :)
Are you going Man hackathon in November @simonjayhawkins ?

**Review by simonjayhawkins — COMMENTED** [review]
@tolaa001 lgtm ping on green

File: pandas/core/arrays/categorical.py (hunk 4 above)
Comment (simonjayhawkins) [reply pandas/core/arrays/categorical.py]: yep.

Comment (tolaa001) [reply pandas/core/arrays/categorical.py]: see you there :)

**Comment by simonjayhawkins** [comment]
Thanks @tolaa001

**Comment by simonjayhawkins** [comment]
is there much of #28724 oustanding?

**Pull request merged**